warehouse/slave.go

```diff
@@ -130,45 +148,45 @@ func (job *Payload) getFileManager(config interface{}, useRudderStorage bool) (f
  * If error occurs with the current config and current revision is different from staging revision
  * We retry with the staging revision config if it is present
  */
-func (jobRun *JobRun) downloadStagingFile(ctx context.Context) error {
-	job := jobRun.job
+func (jr *JobRun) downloadStagingFile(ctx context.Context) error {
+	job := jr.job
 	downloadTask := func(config interface{}, useRudderStorage bool) (err error) {
-		filePath := jobRun.stagingFilePath
+		filePath := jr.stagingFilePath
 		file, err := os.Create(filePath)
 		if err != nil {
 			panic(err)
 		}
 
 		downloader, err := job.getFileManager(config, useRudderStorage)
 		if err != nil {
-			pkgLogger.Errorf("[WH]: Failed to initialize downloader")
+			jr.logger.Errorf("[WH]: Failed to initialize downloader")
 			return err
 		}
 
 		downloadStart := time.Now()
 
 		err = downloader.Download(ctx, file, job.StagingFileLocation)
 		if err != nil {
-			pkgLogger.Errorf("[WH]: Failed to download file")
+			jr.logger.Errorf("[WH]: Failed to download file")
 			return err
 		}
 		file.Close()
-		jobRun.timerStat("download_staging_file_time").Since(downloadStart)
+		jr.timerStat("download_staging_file_time").Since(downloadStart)
```

Comment:
Although of minimal impact, recreating the same metric over and over has a small cost due to the usage of [mutexes](https://github.com/rudderlabs/rudder-go-kit/blob/v0.15.1/stats/otel.go#L329). I suggest we initialize this metrics only once. This one in particular shouldn't even be a `jobRun` one in my opinion, but it can be attached to something else upstream (e.g. like a `warehouseApp` for example :stuck_out_tongue:).

Reply:
Currently, these tags are attached here. 

```GO
func (jr *jobRun) timerStat(name string, extraTags ...Tag) stats.Measurement {
	tags := stats.Tags{
		"module":      moduleName,
		"destType":    jr.job.DestinationType,
		"warehouseID": jr.warehouseID(),
		"workspaceId": jr.job.WorkspaceID,
		"destID":      jr.job.DestinationID,
		"sourceID":    jr.job.SourceID,
	}
	for _, extraTag := range extraTags {
		tags[extraTag.Name] = extraTag.Value
	}
	return jr.stats.NewTaggedStat(name, stats.TimerType, tags)
}
```

Reply:
Not sure whether it can be moved somewhere?

Reply:
Good point.

Reply:
Made the changes.